lib/smack/SplitStructLoadStore.cpp

```diff
@@ -0,0 +1,148 @@
+#include "smack/SplitStructLoadStore.h"
+#include "llvm/IR/InstIterator.h"
+
+namespace smack {
+
+using namespace llvm;
+
+std::vector<Value*> getFirsts(std::vector<std::pair<Value*, unsigned> > lst)
+{
+    std::vector<Value*> ret;
+    for (auto p = lst.begin(); p != lst.end(); ++p)
+      ret.push_back(std::get<0>(*p));
+    return ret;
+}
+
+std::vector<unsigned> getSeconds(std::vector<std::pair<Value*, unsigned> > lst)
+{
+    std::vector<unsigned> ret;
+    for (auto p = lst.begin()+1; p != lst.end(); ++p)
+      ret.push_back(std::get<1>(*p));
+    return ret;
+}
+
+bool SplitStructLoadStore::runOnBasicBlock(BasicBlock& BB)
+{
+  std::vector<Instruction*> toRemove;
+  for(Instruction& I : BB) {
+    if (LoadInst* li = dyn_cast<LoadInst>(&I)) {
+      bool isReturnValue = false;
+      for (auto u = li->user_begin(); u != li->user_end(); ++u)
+      {
+        if (auto ri = dyn_cast<ReturnInst>(*u)) {
+          isReturnValue = true;
+          break;
+        }
+      }
+      if (!isReturnValue && li->getType()->isAggregateType()) {
+        splitStructLoad(li);
```

Comment:
Hmmm, in `splitStructLoad` you do not always perform this splitting since this function start with an if condition. And yet here you always add this instruction into `toRemove`. Is this the right thing to do even when line 58 evaluates to false?